pilot/pkg/xds/delta.go

```diff
@@ -503,7 +503,8 @@ func (s *DiscoveryServer) pushDeltaXds(con *Connection,
 		deltaLog.Debugf("ADS:%v REMOVE for node:%s %v", v3.GetShortType(w.TypeUrl), con.conID, resp.RemovedResources)
 	}
 	// normally wildcard xds `subscribe` is always nil, just in case there are some extended type not handled correctly.
-	if req.Delta.Subscribed == nil && isWildcardResource(w) {
+	// Even wildcard address subscribe, we only push the delta updated resources, set it here have no benefit just makes confusion.
+	if req.Delta.Subscribed == nil && w.TypeUrl != v3.AddressType && isWildcardResource(w) {
```

Comment:
Not sure I understand. `isWildcardResource` already special cases `AddressType`. 

So if we do change it, we should change `isWildcardResource`. But not sure why are we changing?

Reply:
>So if we do change it, we should change isWildcardResource. But not sure why are we changing?

I did think about this, concerned this function can be shared later.

This is to not set w.resourceNames for AddressType here. We do that in underlying generator. 

Setting it here, we can see the resourcesNames change frequently after each push.

As I described [above](https://github.com/istio/istio/pull/46882#issue-1885777799), we can not correctly calculate the removed addresses when a pod/svc delete is handled simultaneously with a network update(a full push, so that when pushRequests merge the config updates are cleared).